src/core/unit.c

```diff
@@ -1742,6 +1745,23 @@ int unit_load(Unit *u) {
                 unit_invalidate_cgroup_members_masks(u);
         }
 
+        /* For top-level cgroups, if not set during loading, restrict allowed CPUs based on manager value */
+        if(u->manager->defaults.allowed_cpus.allocated > 0 && UNIT_HAS_CGROUP_CONTEXT(u)) {
```

Comment:
Instead of checking for specific names of top level slices, changed this to parse the cgroup path.